### PR TITLE
Added tests for the different paths through the pipeline

### PR DESCRIPTION
**Pull request by jannikseidelQBiC**
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).

## Added test profiles for functionality tests to the CI.yml
To ensure that different settings of the flags are functional after a PR these test profiles were added and integrated into the ci.yml.

### CODE REVIEW

**Comment by jannikseidelQBiC** [comment]
@nf-core-bot fix linting

File: conf/test.config [hunk 1]
```diff
@@ -20,13 +20,13 @@ params {
     max_time   = '6.h'
 
     // Input data
-    input  = "${projectDir}/assets/samplesheet.csv"
+    input               = "https://github.com/nf-core/test-datasets/raw/detaxizer/samplesheets/samplesheet.csv"
 
     // Genome references
-    fasta               = "https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/genomics/homo_sapiens/genome/genome.fasta"
+    fasta               = "https://github.com/nf-core/test-datasets/raw/detaxizer/host_reference/genome.hg38.chr21_10000bp_region.fa"
```
Comment (d4straub) [inline conf/test.config:26]: I dont know right now why, but I use `https://raw.githubusercontent.com/nf-core/test-datasets`

**Review by d4straub — COMMENTED** [review]
I am not certain about that github links, could you research that?

File: conf/test.config (hunk 1 above)
Comment (jannikseidelQBiC) [reply conf/test.config:26]: Both formats seem to work, but `https://raw.githubusercontent.com/` looks nicer, changing that in the next commit

File: conf/test_full.config [hunk 2]
```diff
@@ -14,14 +14,13 @@ params {
...
-    // TODO nf-core: Give any required params for the test so that command line flags are not needed
-    input               = "${projectDir}/assets/samplesheet.csv"
+    input               = "https://raw.githubusercontent.com/nf-core/test-datasets/detaxizer/samplesheets/samplesheet.full.csv"
 
     // Genome references
     fasta               = "s3://ngi-igenomes/igenomes/Homo_sapiens/NCBI/GRCh38/Sequence/WholeGenomeFasta/genome.fa"
+
     // Kraken2 test db
-    kraken2db           = "https://genome-idx.s3.amazonaws.com/kraken/k2_standard_08gb_20231009.tar.gz"
+    kraken2db           = "https://zenodo.org/records/8339732/files/k2_HPRC_20230810.tar.gz"
```
Comment (d4straub) [inline conf/test_full.config:23]: Is that file large? If yes, that might be problematic. Using the S3 files is more reliable with test_full.

**Review by d4straub — APPROVED** [review]
LGTM, have just 2 tiny comments

**Pull request merged**